spec/outputs/s3_spec.rb

```diff
@@ -229,4 +229,30 @@
       end
     end
   end
+
+  context "remove empty dirs" do
+
+    before do
+      allow(File).to receive(:directory?).with('/path/to/bar-empty').and_return(true)
+      allow(File).to receive(:directory?).with('/path/to/foo/empty-dir').and_return(true)
+      allow(File).to receive(:directory?).with('/path/to/foo-file').and_return(true)
+      allow(File).to receive(:directory?).with('/path/to/foo-file/file.tmp').and_return(false)
+
+      allow(Dir).to receive(:entries).with('/path/to/bar-empty').and_return([])
+      allow(Dir).to receive(:entries).with('/path/to/foo/empty-dir').and_return(%w(. ..))
+      allow(Dir).to receive(:entries).with('/path/to/foo-file').and_return(%w(/path/to/foo-file/file.tmp))
```

Comment:
Missing `path/to/foo` from mocked case

```suggestion
      allow(Dir).to receive(:entries).with('/path/to/foo-file').and_return(%w(/path/to/foo-file/file.tmp))
      allow(Dir).to receive(:entries).with('/path/to/foo').and_return(%w(. .. empty-dir))
```